source/initial_composition/slab_model.cc

```diff
@@ -0,0 +1,137 @@
+/*
+  Copyright (C) 2023 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <aspect/global.h>
+#include <aspect/initial_composition/slab_model.h>
+#include <aspect/utilities.h>
+#include <aspect/geometry_model/interface.h>
+
+
+namespace aspect
+{
+  namespace InitialComposition
+  {
+    template <int dim>
+    void
+    SlabModel<dim>::initialize ()
+    {
+      AssertThrow (this->get_geometry_model().natural_coordinate_system() == Utilities::Coordinates::spherical,
+                   ExcMessage ("This initial composition plugin can only be used when the "
+                               "preferred coordinate system of the geometry model is spherical "
+                               "(e.g. spherical shell, chunk, sphere)."));
+
+      // The input slabs are defined from the surface of the model
+      surface_boundary_id = this->get_geometry_model().translate_symbolic_boundary_name_to_id("top");
+
+      // The two columns correspond to slabs depth and thickness
+      slab_boundary.initialize({surface_boundary_id}, 2);
+    }
+
+
+    template <int dim>
+    double
+    SlabModel<dim>::
+    initial_composition (const Point<dim> &position,
+                         const unsigned int compositional_index) const
+    {
+      if (compositional_index != slab_index)
```

Comment:
Sorry if this is somehow obvious, but: does this actually work? I don't see where the slab index is being set. 
In a previous version, there was a line
`const unsigned int slab_index          = this->introspection().compositional_index_for_name("slabs");`
and there was also an assert to make sure the slab index exists.